system/lib/pthread/emscripten_futex_wait.c

```diff
@@ -140,7 +140,21 @@ int emscripten_futex_wait(volatile void *addr, uint32_t val, double max_wait_ms)
   if (max_wait_ms != INFINITY) {
     max_wait_ns = (int64_t)(max_wait_ms*1000*1000);
   }
+#ifdef __PIC__
+#define MAX_WAIT (1000*1000*100)
+  // If dynamic linking is supported then we only wait for MAX_WAIT nanoseconds
+  // at a time so we can respond to dlopen calls that might happen on other
+  // threads.
+  int64_t total = 0;
+  while (max_wait_ns == -1 || total < max_wait_ns) {
+    ret = __builtin_wasm_memory_atomic_wait32((int*)addr, val, MAX_WAIT);
+    if (ret != 2) break;
+    _emscripten_yield();
+    total += MAX_WAIT;
```

Comment:
This looks like it can end up waiting more than the requested number of milliseconds in two ways:

1. the wait duration is set to a constant 100 msecs, so if on `emscripten_futex_wait`s for, say, 110 msecs, one might end up getting a wait of 200 msecs (a multiple of 100 msecs)
2. _emscripten_yield() can take up a lot of time(?), which is not accounted into the wait period.

Not sure if msec-accurate waiting is important for `emscripten_futex_wait()` though..

Reply:
True, true.  I will adjust accordingly.

Reply:
I fixed the first issue, by waiting in chunks of 100, but less for the final chunk.   I'm not sure it worth fixing the second issue since it would involve calling out of JS to to get the time each time around the loop.